core/src/main/scala/filodb.core/store/ChunkSink.scala

```diff
@@ -13,7 +13,18 @@ import monix.reactive.Observable
 
 import filodb.core._
 
-case class PartKeyRecord(partKey: Array[Byte], startTime: Long, endTime: Long, hash: Option[Int])
+case class PartKeyRecord(partKey: Array[Byte], startTime: Long, endTime: Long, hash: Option[Int]) {
+  override def equals(obj: Any): Boolean = {
```

Comment:
why do we need to override these?

Reply:
We needed to override equals in `PartKeyRecord` because - 
1. When you do `scanPartKeysByStartTime` and `scanPartKeysByEndTime`, you get duplicates.
1. To get rid of duplicates we put them in a Set.
1. The default equals method that comes with the PartKeyRecord caseClass does not work because dsPartKey inside it is an array which is compared by reference. So the comparison fails even for two identical part key records